IBM/ACE-RISCV
# Promotion to TVM from VMM

## Description of the changes
This PR adds support for TVM promotion initiated by VMM (kvmtool). 

## Type of change

## How to test this PR?
The build process automatically builds a patched version of kvmtool and includes it in the hypervisor's rootfs. The build also create TAP to test local attestation.

Run the hypervisor and then start a TVM using kvmtool
```
./run_linux_vm_kvmtool.sh
```

You should see output similar to that one:
```
# lkvm run -k /root/linux_vm/Image -m 320 -c 2 --name guest-116
# Warning: The maximum recommended amount of VCPUs is 1
#ACE: Promoting a VM to a confidential VM
#ACE: Reading flatten device tree (FDT) at memory address 0x81c00000
#ACE: Virtual machine's memory layout: ConfidentialVmMemoryLayout { kernel: (2147483648, 2483027968), fdt: (2176843776, 2176846417), initrd: None }
#ACE: Reading TEE attestation payload (TAP) at memory address 0x81704000
#ACE: VM measurements: 
PCR0=000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
PCR1=000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
PCR2=000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
PCR3=000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
PCR4=77b9a1c7161f5fcabd409d7b4bea4cbb8cf8a2497323b2371d279657cbbd4de990bdea7f47c520cd6ffbd60f2d62682e
PCR5=000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
PCR6=a14f1ff3eb9b41e94669a80d1aa15143aee724dc32931ad3ceb18e3239cab6ed694f72d6f445cc72b9274e6f5c5c1c63
PCR7=dd530ee7b194f11c9468387baac4249dd573e6f83092444635b0a0d2341f3a29eb5764b7657464a4a0b2a68dbed33ae6
#ACE: Reference PCR4=Sha512=0x77b9a1c7161f5fcabd409d7b4bea4cbb8cf8a2497323b2371d279657cbbd4de990bdea7f47c520cd6ffbd60f2d62682e
#ACE: Attestation succeeded, fetched 1 secrets
#ACE: Created new confidential VM[id=ConfidentialVmId(0)]
```
